util/fipstools/acvp/modulewrapper/modulewrapper.cc

```diff
@@ -2155,6 +2175,34 @@ static bool HKDF(const Span<const uint8_t> args[], ReplyCallback write_reply) {
   return write_reply({Span<const uint8_t>(out_key)});
 }
 
+template <const EVP_MD *(MDFunc)()>
+static bool SSHKDF(const Span<const uint8_t> args[], ReplyCallback write_reply) {
+  const Span<const uint8_t> key = args[0];
+  const Span<const uint8_t> hash = args[1];
+  const Span<const uint8_t> session_id = args[2];
+  const Span<const uint8_t> type = args[3];
+  const Span<const uint8_t> out_bytes = args[4];
+  const EVP_MD *md = MDFunc();
+
+
+  char type_char;
+  memcpy(&type_char, type.data(), sizeof(type_char));
+  unsigned int out_bytes_uint;
+  memcpy(&out_bytes_uint, out_bytes.data(), sizeof(out_bytes_uint));
```

Comment:
Why did you copy just these two? Can you just use type.data and out_bytes.data down below, or should you copy all of them?

Reply:
The function in `openssl/sshkdf.h` requires those types, and implicit casting as well as using `reinterpret_cast` causes the build to fail, so this was my solution. I believe there is precedent in other parts of the file where something similar is done as well.